include/swift/AST/DiagnosticsSIL.def

```diff
@@ -374,6 +374,10 @@ NOTE(autodiff_function_indirect_params_or_result_unsupported,none,
 NOTE(autodiff_external_nondifferentiable_function,none,
      "cannot differentiate an external function that has not been marked "
      "'@differentiable'", ())
+NOTE(autodiff_nondifferentiable_parameter,none,
+     "cannot differentiate through a non differentiable parameter. Consider using .withoutDerivative() to stop gradients.", ())
```

Comment:
* In concrete code, those being passed should be called "arguments" instead of "parameters".
* In diagnostics we should talk about "derivatives" instead of "gradients" since we consider derivatives as a more user-friendly general concept.
* To break lines in diagnostics, use semicolon.
* Use single quotes to wrap around code in diagnostics.

Suggested change:
```
cannot differentiate through a non-differentiable parameter; would you like to add '.withoutDerivative()'?
```

Reply:
I don't think this comment was addressed.